libraries/botframework-connector/src/auth/botFrameworkClientImpl.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import axios from 'axios';
+import { Activity, ChannelAccount, InvokeResponse, RoleTypes } from 'botframework-schema';
+import { BotFrameworkClient } from '../skills';
+import { ServiceClientCredentialsFactory } from './serviceClientCredentialsFactory';
+import { USER_AGENT } from './connectorFactoryImpl';
+import { WebResource } from '@azure/ms-rest-js';
+import { assert } from 'botbuilder-stdlib';
+
+const botFrameworkClientFetchImpl = async (input: RequestInfo, init?: RequestInit): Promise<Response> => {
```

Comment:
In my branch, I went ahead with a full `fetch` implementation. I think we'll need to reconcile our changes such that we support both `axios` and `cross-fetch` in a straightforward way.

Reply:
We could rationalize about it as the samples expressly tell developers to instantiate a [SkillHttpClient to pass in](https://github.com/microsoft/BotBuilder-Samples/blob/560994a556824a45b79f434d782b9804e8b74795/samples/javascript_nodejs/81.skills-skilldialog/dialogRootBot/index.js#L130-L131) (via sample [81.skills-skillDialog](https://github.com/microsoft/BotBuilder-Samples/tree/main/samples/javascript_nodejs/81.skills-skilldialog)).

The behavior from implementation's behavior around [BotFrameworkAuthentication.createBotFrameworkClient()](https://github.com/microsoft/botbuilder-js/blob/cb74e856dab0f81e0943694c8c9b1d8bfafbc781/libraries/botframework-connector/src/auth/botFrameworkAuthentication.ts#L61-L62) could be viewed as a separate, new flow and therefore utilizing `cross-fetch` here is semver acceptable.

What do you think?

Reply:
After discussing offline, we've decided to stay with the current axios-based `fetch` implementation to maintain behavior for developers. Especially given the fact that we'll likely move the samples to utilize the adaptive runtime, which will utilize the `BotFrameworkClientImpl` class.

Reply:
See https://github.com/microsoft/BotBuilder-Samples/issues/3287 for discussion around C# bots using `AddBotRuntime()`.

Resolving conversation.